bluetooth/src/commonMain/kotlin/scanner/Scanner.kt

```diff
@@ -304,20 +343,39 @@ abstract class BaseScanner constructor(
         monitoringPermissionsJob = null
     }
 
-    final override suspend fun scanForDevices(filter: Set<UUID>) {
+    final override suspend fun scanForDevices(filter: Filter, connectionSettings: ConnectionSettings?) {
         if (filter.isEmpty()) {
             logger.info(LOG_TAG) { "Start Scanning" }
         } else {
             logger.info(LOG_TAG) { "Start scanning with filter [${filter.joinToString(", ") { it.uuidString }}]" }
         }
-        didStartScanning(filter)
+        isScanningDevicesMutex.withLock {
+            val dispatcher = singleThreadDispatcher("Scanning for Devices").also {
```

Comment:
If we have one global lazy dispatcher (as is the current pattern) it does not need to be closed. I don't see why this one should be different to be honest.

With coroutines 1.7 it will even be worth considering just switching to `Dispatchers.Default` as it will be multithreaded on native but at least there should be the option to do that.

Reply:
Guess so. Updated